src/main/java/com/targus/ui/WSNPrototype.java

```diff
@@ -0,0 +1,196 @@
+package com.targus.ui;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.targus.utils.Constants;
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.geometry.Point2D;
+import javafx.stage.FileChooser;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class WSNPrototype {
+    private int paneWidth;
+    private int paneHeight;
+    private final ArrayList<Point2D> potentialPositions = new ArrayList<Point2D>();
+
+    public ArrayList<Point2D> getPotentialPositions() {
+        return potentialPositions;
+    }
+
+    private final ArrayList<Point2D> targets = new ArrayList<Point2D>();
+
+    public ArrayList<Point2D> getTargets() {
+        return targets;
+    }
+
+    private final IntegerProperty mProperty = new SimpleIntegerProperty(1);
+
+    public IntegerProperty getMProperty() {
+        return mProperty;
+    }
+
+    private final IntegerProperty kProperty = new SimpleIntegerProperty(1);
+
+    public IntegerProperty getKProperty() {
+        return kProperty;
+    }
+
+    private final DoubleProperty communicationRangeProperty = new SimpleDoubleProperty(100);
```

Comment:
In the `Constants` class, we can define `static final` variables for the magic numbers. After we define this variable, we can replace 100 with `Constants.DEFAULT_COMMUNICATION_RANGE`.

Also, we cannot alter the value of the communication range, at least I could not see a setter method. In the previous implementation, we could change this value. We need to define a setter method for this member and for all the other members